### DISCUSSION TITLE
Alternative ways to extract point time series from mHM output without adjusting idgauges grid.

### DISCUSSION THREAD

**Discussion by pallavhydro** [mhm-ufz/mHM#58] [Ideas]
The 8th mHM user forum saw a request on alternative ways to extract point time series from mHM output without adjusting idgauges grid.

### Background

mHM's inbuilt feature to extract point discharge simulation time series is based on `location` information from `idgauges grid` and  using the `gauge id` from `evaluation_gauges` section of `mhm.nml`.

This feature has two constraints -
1. This is only useful for extracting streamflow and not available for other model variables.
2. idgauges grid needs to be updated in case data extraction is required at more points than those incorporated.

### Solution

The alternatives to extracting point time series data from mHM output grid files would be -
1. Post-processor scripts (in R, Python, etc.) for extraction purpose
2. Using netcdf specialist platforms such as CDO, NCO . Examples using CDO are given below -

**One liner samples** for extracting data at `Lat = 27.5 deg` and `Lon = 85.33 deg.` using CDO would be -

```
# To extract depth averaged soil moisture from mHM output netcdf file to soilmoisture_timeseries.txt:
cdo -L outputtab,date,value -remapnn,lon=85.33_lat=27.5 -selvar,SM_Lall mHM_Fluxes_States.nc > soilmoisture_timeseries.txt

# To extract streamflow from mRM output netcdf file to streamflow_timeseries.txt:
cdo -L outputtab,date,value -remapnn,lon=85.33_lat=27.5 -selvar,Qrouted mRM_Fluxes_States.nc > streamflow_timeseries.txt
```

An example output is as below.
[soilmoisture_timeseries.txt](https://github.com/mhm-ufz/mHM/files/9728691/soilmoisture_timeseries.txt)

Note: CDO can be installed rather easily -
- In MacOS, using home brew: `brew install cdo`.
- In Linux (e.g. Ubuntu OS). using apt-get: `sudo apt-get install cdo`.
A more comprehensive note on installation of CDO can be found in [this page](https://www.isimip.org/protocol/preparing-simulation-files/cdo-help/)

**Comment by vdungnguyen** [comment]
Thanks @pallavhydro for the suggestion. That is actually the method (with cdo) I use to extract the time series from the 'Fluxes_States' files, particularly when the files are so large to handle in R or Python (from 5-40Gb). I wonder if it would be good to write (by adding a few more Fortran code lines) these point series at the writing output data stage of the mHM as long as the list of points (coordinates) is given. Because, with this we may not need to write such big files.
